src/pm-conversations/pmConversationsModel.js

```diff
@@ -190,7 +194,15 @@ export function reducer(
       // TODO optimize; this is quadratic (but so is the webapp's version?!)
       let st = initialState;
       for (const r of action.data.recent_private_conversations ?? []) {
-        st = insert(st, keyOfExactUsers(r.user_ids), r.max_message_id);
+        st = insert(
+          st,
+          keyOfExactUsers(
+            r.user_ids
+              // Copy the array (keyOfExactUsers sorts its input in-place)
+              .map(id => id),
+          ),
+          r.max_message_id,
+        );
```

Comment:
Hmm, awkward that this makes this one line sprawl across 9 lines. I'd take that as a cue to pull out some subexpression as a local :wink: 

E.g. like this:
```diff
       for (const r of action.data.recent_private_conversations ?? []) {
-        st = insert(
-          st,
-          keyOfExactUsers(
-            r.user_ids
-              // Copy the array (keyOfExactUsers sorts its input in-place)
-              .map(id => id),
-          ),
-          r.max_message_id,
-        );
+        const key = keyOfExactUsers([...r.user_ids]);
+        st = insert(st, key, r.max_message_id);
       }
```

I think the comment isn't actually needed, because after all the types say we need a mutable array and that we have a read-only one. This bit of code doesn't actually need to care about *why* the helper demands a mutable array.

(Meanwhile although the `map` works fine, the spread feels more direct to me, and also seems more likely to be reliably efficient -- no callback to invoke for each element.)